contracts/script/fp/DeployOPSuccinctDG.s.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {Script} from "forge-std/Script.sol";
+import {console} from "forge-std/console.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {DisputeGameFactory} from "src/dispute/DisputeGameFactory.sol";
+import {OPSuccinctFaultDisputeGame} from "src/fp/OPSuccinctFaultDisputeGame.sol";
+import {SP1MockVerifier} from "@sp1-contracts/src/SP1MockVerifier.sol";
+import {Claim, GameType, Hash, OutputRoot, Duration} from "src/dispute/lib/Types.sol";
+import {IDisputeGame} from "src/dispute/interfaces/IDisputeGame.sol";
+import {IDisputeGameFactory} from "src/dispute/interfaces/IDisputeGameFactory.sol";
+import {ISP1Verifier} from "@sp1-contracts/src/ISP1Verifier.sol";
+
+contract DeployOPSuccinctDG is Script {
+    function run() public {
+        vm.startBroadcast();
+
+        // Deploy the factory implementation
+        DisputeGameFactory factoryImpl = new DisputeGameFactory();
+
+        // Deploy factory proxy
+        ERC1967Proxy factoryProxy = new ERC1967Proxy(
+            address(factoryImpl), abi.encodeWithSelector(DisputeGameFactory.initialize.selector, msg.sender)
+        );
+        DisputeGameFactory factory = DisputeGameFactory(address(factoryProxy));
+
+        // Setup starting anchor roots
+        GameType gameType = GameType.wrap(uint32(vm.envUint("GAME_TYPE")));
+
+        // Get or deploy SP1 verifier based on environment variable
+        address sp1VerifierAddress;
+        bool useMockVerifier = vm.envOr("USE_SP1_MOCK_VERIFIER", false);
+
+        if (useMockVerifier) {
+            // Deploy mock verifier for testing
+            SP1MockVerifier sp1Verifier = new SP1MockVerifier();
+            sp1VerifierAddress = address(sp1Verifier);
+            console.log("Using SP1 Mock Verifier:", address(sp1Verifier));
+        } else {
+            // Use provided verifier address for production
+            sp1VerifierAddress = vm.envAddress("SP1_VERIFIER_GATEWAY");
```

Comment:
nit: Change this to `VERIFIER_ADDRESS` to match the other contract deployment scripts.